test/dogear.test.js

```diff
@@ -243,82 +240,82 @@ describe('Plugin reports on paths.', (tap) => {
 
 describe('Plugin reports ops.', (tap) => {
 
-  tap.test('Starts listening to ops reports.', (t) => {
+  tap.test('Starts listening to ops reports.', async t => {
     t.plan(1);
 
-    const server = initServer();
+    const server = await initServer();
 
-    server.register({
-      register: plugin,
+    await server.register({
+      plugin,
       options: {
         statsdConfig: {
           mock: true
         },
         opsInterval: 500
       }
-    }).then(() => {
+    });
 
-      const plugin = server.plugins['dogear'];
+    const dogear = server.plugins['dogear'];
 
-      t.ok(plugin._ops);
+    t.ok(dogear._ops);
 
-      server.stop({ timeout: 0 }).then(t.end);
-    });
+    await server.stop({ timeout: 0 });
+    t.end();
   });
 
 
-  tap.test('Report ops metrics.', (t) => {
-    t.plan(2);
+  tap.test('Report ops metrics.', async t => {
+    t.plan(3);
 
-    const server = initServer();
+    const server = await initServer();
 
-    server.register({
-      register: plugin,
+    await server.register({
+      plugin,
       options: {
         statsdConfig: {
           mock: true
         },
-        opsInterval: 500
+        opsInterval: 10
       }
-    })
-    .then(() => {
+    });
 
-      const client = server.statsd;
+    const client = server.statsd;
+    const gaugeSpy = sandbox.spy(client, 'gauge');
```

Comment:
I believe a stub should be safer here to prevent the client from attempting to do work. We just want to verify the gauge was invoked.

Reply:
Sinon stubs with the signature `sinon.stub(object, "method", func)` have been deprecated in v3.0.0 of sinon. I also tried `stub.callsFake(fakeFunction)` but this won't work with async function. And the only thing we want to achieve here, is to verify the gauge was invoked (as you mentioned). We don't need to replace the whole function for this. Btw: Where should the client send anything when there's no statsd server speciefied anywhere in the whole repository.